### DISCUSSION TITLE
Conditionally register navigations and group them as desired

### DISCUSSION THREAD

**Discussion by romaldyminaya** [filamentphp/filament#1701] [Help] [answered]
I'd like to know how register navigations based on a higher level than navigationGroup. For example

PersonResource.php => Group (Contact) ---------- |
AddressResource.php => Group (Contact)---------  |  Contacts Module
CountryResource.php => Group (Configuration)--   |

In the example above I have a module called Contacts Module, and I'd like that whether I access any of the resources inside that module,  that each resource register their corresponding navigation items. So I can take advantage of the navigationGroup.

I'm wondering also, how can I get the current resource or page or module called from any resource or page ?

**Comment by titantwentyone** [comment]
Could you post some code and we can take a look?

**Reply by romaldyminaya** [reply]
I found a work around, its kind of messy but it works, in every request I have to get the Route beeing requested by the user, with that, I get the class beeing called and then that class has the public static property I need to access.

Here is the code that worked for me:

```
namespace App\Filament\Apps;

use ReflectionClass;
use Filament\Facades\Filament;
use Filament\Resources\Resource;
use Illuminate\Support\Facades\Route;
use App\Interfaces\Filament\ResourceApp;

abstract class BaseApp extends Resource implements ResourceApp
{
    abstract public static function getAppName(): string;
    abstract public static function getDefaultUrl(): string;

    public static function registerNavigationItems(): void
    {
        if (is_subclass_of(Route::current()->controller, Resource::class)) {
            return;
        }

        if (method_exists(new (Route::current()->controller), 'getResource')) {
            $appName = Route::current()->controller->getResource()::getAppName();

            if ($appName != static::getAppName()) {
                return;
            }
        }

        if (!static::shouldRegisterNavigation()) {
            return;
        }

        if (!static::canViewAny()) {
            return;
        }

        Filament::registerNavigationItems(static::getNavigationItems());
    }
}
```

With that abstract class, I just need to extend in every controller and implement the getAppName and AppDefaultUrl.

With that I achieved my goal.

**Comment by danharrin** [comment, marked as answer]
No, at the moment sidebar navigation only goes 2 levels deep - groups & items.
